AGENTS.md

```diff
@@ -0,0 +1,443 @@
+
+**Azure Developer CLI:**
+- `azure.yaml` - Defines azd hooks (postprovision scripts)
+
+**Environment Variables:**
+- `.env.sample` - Example .env file showing all possible configurations
+- `.env.sample.azure` - Azure-specific example
+- `.env.sample.github` - GitHub Models example
+- `.env.sample.ollama` - Ollama example
+- `.env.sample.openai` - OpenAI.com example
+
+### GitHub Workflows (.github/workflows/)
+
+**`python.yaml` - Linting and Formatting Check:**
+- Runs on: push to main, pull requests to main
+- Uses: uv for virtual environment and dependency installation
+- Runs: `uv run ruff check .` and `uv run black . --check --verbose`
+- **Important:** The CI uses `uv` but local development typically uses standard `pip`
+
+**`test-github-models.yaml` - Integration Test:**
+- Runs on: push to main, pull requests to main (limited paths)
+- Tests: `chat.py` and `spanish/chat.py` with GitHub Models
+- Uses: uv for setup, requires models: read permission
+- Sets: `API_HOST=github`, `GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}`, `GITHUB_MODEL=openai/gpt-4o-mini`
+
+### Dev Container Files (.devcontainer/)
+
+- `.devcontainer/devcontainer.json` - Default dev container (Azure OpenAI setup with azd)
+- `.devcontainer/Dockerfile` - Base Python 3.12 image, installs all requirements-dev.txt
+- `.devcontainer/github/` - GitHub Models variant
+- `.devcontainer/ollama/` - Ollama variant  
+- `.devcontainer/openai/` - OpenAI.com variant
+
+All dev containers install all dependencies from `requirements-dev.txt` which includes base, RAG, and dev tools.
+
+## Running the Code
+
+### Environment Setup
+
+**ALWAYS follow these steps in order:**
+
+1. **Create a Python virtual environment** (if not in a Dev Container or Codespaces):
+   ```bash
+   python -m venv .venv
+   source .venv/bin/activate  # On Windows: .venv\Scripts\activate
+   ```
+
+2. **Install base dependencies:**
+   ```bash
+   python -m pip install -r requirements.txt
+   ```
+
+3. **For RAG examples, ALSO install RAG dependencies:**
+   ```bash
+   python -m pip install -r requirements-rag.txt
+   ```
+
+4. **For development (linting, formatting), install dev dependencies:**
+   ```bash
+   python -m pip install -r requirements-dev.txt
+   ```
+   Note: `requirements-dev.txt` includes both `requirements.txt` and `requirements-rag.txt` via `-r` directives.
+
+### Configuring LLM Provider
+
+**For agents, ALWAYS prefer GitHub Models** since the agent often won't have access to Azure OpenAI or paid API keys.
+
+The scripts read environment variables from a `.env` file. Create one based on your provider:
+
+#### Option 1: GitHub Models (RECOMMENDED for agents)
+
+In GitHub Codespaces, `GITHUB_TOKEN` is already set, so **no .env file is needed** - scripts will work immediately.
+
+For local development:
+1. Create a GitHub Personal Access Token (no scopes needed): https://github.com/settings/tokens
```

Comment:
Agents cant create a GITHUB PAT, but sometimes agent work in environment where theyre available. They can check to see if GITHUB_TOKEN is an env var.